crate/cli/src/actions/verify.rs

```diff
@@ -0,0 +1,93 @@
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use cosmian_kms_utils::tee::forge_report_data;
+use openssl::x509::X509;
+use rand::Rng;
+use ratls::verify::get_server_certificate;
+use tee_attestation::verify_quote;
+use tokio::task::spawn_blocking;
+
+use crate::{
+    config::CliConf,
+    error::{result::CliResultHelper, CliError},
+};
+
+/// Query the KMS to check its trustworthiness. Validate the TLS certificate to use in any further queries.
+#[derive(Parser, Debug)]
+pub struct TeeAction {
```

Comment:
Actually it's well-known for people working on TEE :D Otherwise no. But there is no good word currently to designed secure environments such as SEV or SGX. Secure VM, Enclave, etc. are technology dependent.